### ISSUE TITLE
[BUG] Function update does not work if model consists of several Schemas after 2.5.0 version.

### ISSUE THREAD

**Issue by DedWatson** [dynamoose/dynamoose#1132]
### Summary:
Function **update** does not work if model consists of several Schemas after 2.5.0 version.

#### Schema
```javascript
const schemaOne = new dynamoose.Schema({
    id: {
        type: String,
        hashKey: true
    },
    rangeKey: {
        type: String,
        rangeKey: true
    },
    catData: {
        type: Object,
        schema: {
            parameter: String
        }
    }
}, {timestamps: true});

const schemaTwo = new dynamoose.Schema({
    id: {
        type: String,
        hashKey: true
    },
    rangeKey: {
        type: String,
        rangeKey: true
    },
    dogData: {
        type: Object,
        schema: {
            anotherParameter: String
        }
    },
}, {timestamps: true});
```

#### Model
```javascript
const test = dynamoose.model("Test", [schemaOne, schemaTwo]);
```

#### General
```javascript
test.create({id: "test", rangeKey: "range"}).then((value => {
    test.update({id: "test", rangeKey: "range"}, {
        dogData: {
            "anotherParameter": "some value"
        }
    }).then((value => {
        console.log(value);
    }))
}))
```

### Current output and behavior (including stack trace):
**dynamoose@2.7.0**

```
aws:dynamodb:describeTable:request - {
    "TableName": "Test"
}
aws:dynamodb:describeTable:response - {
    "Table": {
        "AttributeDefinitions": [
            {
                "AttributeName": "id",
                "AttributeType": "S"
            },
            {
                "AttributeName": "rangeKey",
                "AttributeType": "S"
            }
        ],
        "TableName": "Test",
        "KeySchema": [
            {
                "AttributeName": "id",
                "KeyType": "HASH"
            },
            {
                "AttributeName": "rangeKey",
                "KeyType": "RANGE"
            }
        ],
        "TableStatus": "ACTIVE",
        "CreationDateTime": "2021-03-02T16:18:16.484Z",
        "ProvisionedThroughput": {
            "NumberOfDecreasesToday": 0,
            "ReadCapacityUnits": 1,
            "WriteCapacityUnits": 1
        },
        "TableSizeBytes": 0,
        "ItemCount": 0,
        "TableArn": "arn:aws:dynamodb:ap-southeast-2:685385005121:table/Test",
        "TableId": "130bddf7-5e08-450e-a972-75ef384e10f3"
    }
}
aws:dynamodb:putItem:request - {
    "Item": {
        "id": {
            "S": "test"
        },
        "rangeKey": {
            "S": "range"
        },
        "createdAt": {
            "N": "1614702101672"
        },
        "updatedAt": {
            "N": "1614702101672"
        }
    },
    "TableName": "Test",
    "ConditionExpression": "attribute_not_exists(#__hash_key)",
    "ExpressionAttributeNames": {
        "#__hash_key": "id"
    }
}
aws:dynamodb:putItem:response - {}
aws:dynamodb:updateItem:request - {
    "Key": {
        "id": {
            "S": "test"
        },
        "rangeKey": {
            "S": "range"
        }
    },
    "ReturnValues": "ALL_NEW",
    "ExpressionAttributeNames": {
        "#a0": "updatedAt"
    },
    "ExpressionAttributeValues": {
        ":v0": {
            "N": "1614702104627"
        }
    },
    "UpdateExpression": "SET #a0 = :v0",
    "TableName": "Test"
}
aws:dynamodb:updateItem:response - {
    "Attributes": {
        "createdAt": {
            "N": "1614702101672"
        },
        "id": {
            "S": "test"
        },
        "rangeKey": {
            "S": "range"
        },
        "updatedAt": {
            "N": "1614702104627"
        }
    }
}
Document {
  createdAt: 2021-03-02T16:21:41.672Z,
  id: 'test',
  rangeKey: 'range',
  updatedAt: 2021-03-02T16:21:44.627Z
}

```

### Expected output and behavior:

**dynamoose@2.5.0**

```
aws:dynamodb:describeTable:request - {
    "TableName": "Test"
}
aws:dynamodb:describeTable:response - {
    "Table": {
        "AttributeDefinitions": [
            {
                "AttributeName": "id",
                "AttributeType": "S"
            },
            {
                "AttributeName": "rangeKey",
                "AttributeType": "S"
            }
        ],
        "TableName": "Test",
        "KeySchema": [
            {
                "AttributeName": "id",
                "KeyType": "HASH"
            },
            {
                "AttributeName": "rangeKey",
                "KeyType": "RANGE"
            }
        ],
        "TableStatus": "ACTIVE",
        "CreationDateTime": "2021-03-02T16:18:16.484Z",
        "ProvisionedThroughput": {
            "NumberOfDecreasesToday": 0,
            "ReadCapacityUnits": 1,
            "WriteCapacityUnits": 1
        },
        "TableSizeBytes": 0,
        "ItemCount": 0,
        "TableArn": "arn:aws:dynamodb:ap-southeast-2:685385005121:table/Test",
        "TableId": "130bddf7-5e08-450e-a972-75ef384e10f3"
    }
}
aws:dynamodb:putItem:request - {
    "Item": {
        "id": {
            "S": "test"
        },
        "rangeKey": {
            "S": "range"
        },
        "createdAt": {
            "N": "1614702514472"
        },
        "updatedAt": {
            "N": "1614702514472"
        }
    },
    "TableName": "Test",
    "ConditionExpression": "attribute_not_exists(#__hash_key)",
    "ExpressionAttributeNames": {
        "#__hash_key": "id"
    }
}
aws:dynamodb:putItem:response - {}
aws:dynamodb:updateItem:request - {
    "Key": {
        "id": {
            "S": "test"
        },
        "rangeKey": {
            "S": "range"
        }
    },
    "ReturnValues": "ALL_NEW",
    "ExpressionAttributeNames": {
        "#a0": "updatedAt",
        "#a1": "dogData"
    },
    "ExpressionAttributeValues": {
        ":v0": {
            "N": "1614702517055"
        },
        ":v1": {
            "M": {
                "anotherParameter": {
                    "S": "some value"
                }
            }
        }
    },
    "UpdateExpression": "SET #a0 = :v0, #a1 = :v1",
    "TableName": "Test"
}
aws:dynamodb:updateItem:response - {
    "Attributes": {
        "createdAt": {
            "N": "1614702514472"
        },
        "dogData": {
            "M": {
                "anotherParameter": {
                    "S": "some value"
                }
            }
        },
        "id": {
            "S": "test"
        },
        "rangeKey": {
            "S": "range"
        },
        "updatedAt": {
            "N": "1614702517055"
        }
    }
}
Document {
  createdAt: 2021-03-02T16:28:34.472Z,
  dogData: { anotherParameter: 'some value' },
  id: 'test',
  rangeKey: 'range',
  updatedAt: 2021-03-02T16:28:37.055Z
}
```

### Environment:

Operating System: MacOS 10.15.6 and Ubuntu 18.04.4
Operating System Version: MacOS 10.15.6 and Ubuntu 18.04.4
Node.js version (`node -v`): v12.18.4
NPM version: (`npm -v`): 7.5.4
Dynamoose version: 2.7.0

### Other information (if applicable):

On Dynamoose 2.7.0 [this function](https://github.com/dynamoose/dynamoose/blob/51401dea54714c563bc9de3924665982cbbc9546/lib/Model/index.ts#L738) return undefined

### Other:
- [x] I have read through the Dynamoose documentation before posting this issue
- [x] I have searched through the GitHub issues (including closed issues) and pull requests to ensure this issue has not already been raised before
- [x] I have searched the internet and Stack Overflow to ensure this issue hasn't been raised or answered before
- [x] I have tested the code provided and am confident it doesn't work as intended
- [x] I have filled out all fields above
- [x] I am running the latest version of Dynamoose

**Comment by fishcharlie** [comment]
@DedWatson First of all. Just want to thank you for this report. The details you provided were extremely helpful. I have used [git bisect](https://git-scm.com/docs/git-bisect) to identify that this functionality broke in the commit 330e8169469f6cda37db191cdd42594f0d771c29. Which was included in PR #1089. A test for that fix was included in commit b1557dfd0e37846ecdeb565aaa51751bf5ad0d3b.

At this point, I'm unsure why that commit broke that functionality for you.

I will make this a high priority tho since we have pinpointed where this issue is occurring and should be easy to dive into figuring out how to fix it.

Hope to have a PR for you to review & test soon. Thanks again for the report.
